src/vsr/journal.zig

```diff
@@ -1643,7 +1643,7 @@ pub fn JournalType(comptime Replica: type, comptime Storage: type) type {
             assert(journal.has_dirty(message.header));
 
             const write = journal.writes.acquire() orelse {
```

Comment:
Just for my understanding:

- it indeed can happen that a node runs out of IOPs, so prepare won't get written
- however, the prepare would still be in memory (pipeline or cache), so we'll retry writing it

?

Reply:
> it indeed can happen that a node runs out of IOPs, so prepare won't get written

Yes

> however, the prepare would still be in memory (pipeline or cache), so we'll retry writing it

Depends if we're a primary or backup:

- If we're a primary and this is not a repair, then yes, it is in the pipeline, and we will try again.
- If we're a primary and this is a repair, then the retry will be driven by us requesting the prepare all over again.
- If we're a backup, whether it is a repair or not, retries are driven by requesting prepares. We do check our cache for prepares before requesting them.